### PR TITLE
Alignment for templated release process 

### PR DESCRIPTION
**Pull request by phoinixgrr**
#### Summary

We introduce all relevant changes, based on the templated release process efforts. 

Specifically: 
- Dockerfile changes
- GitLab Pipeline
- Makefile template

#### Ticket Link

Ticket: https://mattermost.atlassian.net/browse/DOPS-1086
#### Release Note

```release-note

```

### CODE REVIEW

**Comment by phoinixgrr** [comment]
There is some CI enabled using github-actions. 
Will migrate this also

File: Makefile [hunk 1]
```diff
@@ -1,139 +1,360 @@
...
+go-build: $(GO_BUILD_PLATFORMS_ARTIFACTS) ## to build binaries
 
-.PHONY: clean
-## clean: deletes all
-clean:
-	rm -rf build/_output/bin/
-
-.PHONY: dist
-## dist-aws: creates the bundle file for AWS Lambda deployments
-dist: build
```
Comment (ghost) [inline Makefile:89]: Are we sure that lambda deployment bundle is not needed ?

**Comment by phoinixgrr** [comment]
@spirosoik 

We bundle terraform binary and use `exec()` to call it inside ponos. 
Is there a reason for not choosing to use a [Go Library](https://github.com/hashicorp/terraform-exec)  to perform this  ?

File: Makefile (hunk 1 above)
Comment (phoinixgrr) [reply Makefile:89]: good catch @pfltdv  👍 

Will make sure to include it in the templated makefile.

**Comment by spirosoik** [comment]
@phoinixgrr there is no library for this. We do the same for multiple services we have btw.

File: Makefile (hunk 1 above)
Comment (spirosoik) [reply Makefile:89]: I was ready to say that this is definitely needed

Comment (phoinixgrr) [reply Makefile:89]: Pushed https://github.com/mattermost/ponos/pull/2/commits/364bd1a0f1b2f63d2170b661be68147aa0172ec6 to assess this

**Review by ghost — DISMISSED** [review]
:100: Nice work Akis.

**Comment by spirosoik** [comment]
@phoinixgrr one more thing learn and be aware. There is a pipeline in Gitlab where we use to package and push every app like Ponos to s3.

**Review by stafot — DISMISSED** [review]

**Comment by phoinixgrr** [comment]
> @phoinixgrr one more thing learn and be aware. There is a pipeline in Gitlab where we use to package and push every app like Ponos to s3.

Sure, Refactoring this by adding a target to upload bundle to s3 👍 
Will follow with a commit

**Comment by phoinixgrr** [comment]
@spirosoik  Pushed https://github.com/mattermost/ponos/pull/2/commits/371e35b5785a1f40c9bab20eca336df62738cd90 to assess https://github.com/mattermost/ponos/pull/2#issuecomment-1158812127


@stafot @pfltdv dropped your reviews because of that. Do re-review. 🙇

**Review by ghost — APPROVED** [review]
LGTM

**Review by stafot — APPROVED** [review]

**Review by spirosoik — APPROVED** [review]
lgtm

**Pull request merged**